ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/Metric.scala

```diff
@@ -141,13 +194,47 @@ object Metric {
   }
 
   object DelayMetric {
+
+    def empty[T](
+        recordTimeFunction: T => Seq[Timestamp],
+        objectives: List[ServiceLevelObjective[Value]],
+        clock: Clock,
+    ): DelayMetric[T] =
+      DelayMetric(
+        recordTimeFunction = recordTimeFunction,
+        clock = clock,
+        objectives = objectives.map(objective => objective -> None).toMap,
+      )
+
     final case class Value(meanDelaySeconds: Option[Long]) extends MetricValue {
       override def formatted: List[String] =
         List(s"mean delay: ${meanDelaySeconds.getOrElse("-")} [s]")
     }
 
-    def empty[T](recordTimeFunction: T => Seq[Timestamp], clock: Clock): DelayMetric[T] =
-      DelayMetric(recordTimeFunction, clock)
+    object Value {
+      implicit val valueOrdering: Ordering[Value] = (x: Value, y: Value) => {
+        (x.meanDelaySeconds, y.meanDelaySeconds) match {
+          case (Some(xx), Some(yy)) =>
+            if (xx < yy) -1
+            else if (xx > yy) 1
+            else 0
+          case (Some(_), None) => 1
+          case (None, Some(_)) => -1
+          case (None, None) => 0
+        }
+      }
+    }
+
+    final case class MaxDelay(maxDelaySeconds: Long) extends ServiceLevelObjective[Value] {
+      override def isViolatedBy(metricValue: Value): Boolean =
+        metricValue.meanDelaySeconds.exists(_ > maxDelaySeconds)
+
+      override def maxViolatingValue(first: Value, second: Value): Value =
```

Comment:
Why do you need this method still? Can't you inline it where it is used since it doesn't use anyway anything from the `MaxDelay` context?

Reply:
It's just a matter of separation on responsibilities. I cannot convince myself to hardcode `Ordering[Type].max(a,b)` inside a metric because we could potentially have a metric where one objective was for finding a max value, and another for finding a min value.
But as this is only a potential problem that might not even occur, I will inline this to make it more clean just as you suggest.
Thanks!